src/d3a/d3a_core/simulation.py

```diff
@@ -539,6 +540,49 @@ def __setstate__(self, state):
         self.__dict__.update(state)
         self._load_setup_module()
 
+    def loop_area_object(self, target_area):
+        duplicate_values = ""
+        try:
+            if target_area.parent is None:
+                duplicate_values = self.get_duplicate_data(target_area.children)
+                if len(duplicate_values) > 0:
+                    self.rename_area_object(target_area.children, duplicate_values)
+            for child in target_area.children:
+                if len(child.children) > 0:
+                    duplicate_values = self.get_duplicate_data(child.children)
+                if len(duplicate_values) > 0 and len(child.children):
+                    self.rename_area_object(child.children, duplicate_values)
+                    self.loop_area_object(child)
```

Comment:
```suggestion
                if len(child.children) > 0:
                    duplicate_values = self.get_duplicate_data(child.children)
                    if len(duplicate_values) > 0:
                        self.rename_area_object(child.children, duplicate_values)
                    self.loop_area_object(child)
```

Reply:
this should be len(child.children) > 0, made this change

Reply:
You are checking for that already in Line 558

Reply:
actually these are two are seperate if conditions, so i felt better to check, but still if we wont check also its fine, i have made this change